### DISCUSSION TITLE
Organize respositories in different Subfolders/Categories in GitHub Desktop

### DISCUSSION THREAD

**Discussion by A320-Daniel** [community/community#24091] [Apps, API and Webhooks]
<p>Hi Everybody,</p>
<p>I got a question to the community. I have my own git on my webserver with quite a lot of repositories on it.<br>
I just found the GitHub Desktop app a few days ago and I think this is a very nice tool to work with Git.<br>
I can also add my repositories which are cloned on my local harddisk. All are in a Category “Others”. What I would like to do is group some of the repositories to get a better kind of organization.<br>
I try to explain what I mean by using a generic example. Lets say I have repositories 1 to 30. Repo 1 to 10 belongs to a “Project A”, Repo 11 to 20 to “Project B” and Repo 21 to 30 to “Project C”.<br>
Instead of having all 30 Repos in the category “Others” in GitHub I’d like to create a group “Project A” and put repo 1 to 10 in this group and so on.<br>
I hope I could explain what I am talking about :slight_smile:<br>
Thanks for any help<br>
Daniel</p>

**Comment by steveward** [comment, marked as answer]
<p>Welcome <a class="mention" href="/a320-daniel">@A320-Daniel</a>!</p>
<p>GitHub Desktop will categorize any non-GitHub repositories under <code>Other</code>. A non-GitHub repository is classified as a local repository that does not have a remote repository on GitHub or a repository that is hosted on a non-GitHub hosting service (like GitLab).</p>
<p>At this time we do not have the option to create custom repository groups, but this is something that has been previously requested (see <a href="https://github.com/desktop/desktop/issues/4945">this issue</a> for an example and <a href="https://github.com/desktop/desktop/issues/6460">this issue</a> for broader repository list efforts). At this time we don’t have any plans to make changes to repository organization, but this will likely be something we reconsider again in the future. Thanks for the feedback about this.</p>

**Reply by irares** [reply]
I thins this is a must-have feature when we have multiple products and clients and many repos. Why ignoring the feature?

**Reply by rohitkhatri** [reply]
I too believe that It's a must have

**Reply by AtAltitude** [reply]
I also think this is a must-have. On GitHub as well as GitLab I have all my projects nicely organized by which client they're for, what overarching project they're part of, whether they're general purpose-utilities, or if they're personal projects. On the GitHub desktop client, it's a jumbled mess of all projects haphazardly thrown into whether they're on GitHub or not. Those categories are not useful.

I can somewhat work around this by removing repositories that I haven't used in a while locally, but even so I still have multiple projects running in parallel. Not to mention that doing this is adding significant overhead to my workflow, particularly for small maintenance-related changes.

It's becoming a serious usability issue for me. The two categories are fine if you have maybe a dozen projects and see all of them on-screen at once, but for any more than that it becomes increasingly difficult to navigate. The only somewhat saving grace here is the Recents category.

I'd expect a piece of software as mature as GitHub Desktop to have this functionality. I fail to understand how it's possible that this wasn't a priority up to this point.

**Reply by Woudjee** [reply]
I fully agree. I like how simple this is, but key features like these are really a must. I will look for an alternative client.

**Reply by AtAltitude** [reply]
> I fully agree. I like how simple this is, but key features like these are really a must. I will look for an alternative client.

I can recommend Fork. It's not free, but it's a buy-once-use-forever deal and it's a lot more feature-complete than the normal GitHub desktop client. Been using it since shortly after my last post here and it's been great.

**Reply by Woudjee** [reply]
Aha, good to read! I indeed just came across it during my search for alternatives. Does it have the opportunity to categorize repositories per project? In my DevOps environment, I have multiple DevOps projects. Each project houses multiple repositories with different scopes. So it would be a-ma-zing if I have a clear overview in what belongs where.

**Reply by AtAltitude** [reply]
> Aha, good to read! I indeed just came across it during my search for alternatives. Does it have the opportunity to categorize repositories per project? In my DevOps environment, I have multiple DevOps projects. Each project houses multiple repositories with different scopes. So it would be a-ma-zing if I have a clear overview in what belongs where.

I'm not sure if there's a way to do that automatically, but you could easily create a folder structure that has one folder for each project, which then contains all repositories for that project. Those folders can also have sub-folders, so you can go all out on organization.

**Reply by Woudjee** [reply]
That really is amazing. I will try it out tomorrow 😁👍

**Reply by LambdaScorpii** [reply]
Hey @steveward ,

hope you are doing well. 

Are there any news on this regard? It's been 3 years, so I would really like to see this feature. 

I also can't imagine that it is too much trouble to implement, as there obviously is already a categorisation in the code, so you need to make this user accessible... or derive it from a parent folder that it is contained in :) 

Cheers
